client/my-sites/patterns/components/page-view-tracker.tsx

```diff
@@ -61,23 +51,24 @@ export function PatternsPageViewTracker( {
 	}, [ category, isDevAccount, isLoggedIn, patternTypeFilter ] );
 
 	useEffect( () => {
-		if ( isDevAccount !== undefined ) {
+		if ( isDevAccount !== undefined && patternsCount !== undefined ) {
 			recordTracksEvent( 'calypso_pattern_library_view', {
 				category,
 				is_logged_in: isLoggedIn,
 				user_is_dev_account: isDevAccount ? '1' : '0',
-				search_term: debouncedSearchTerm,
+				search_term: searchTerm,
 				type: getTracksPatternType( patternTypeFilter ),
 				view,
 				referrer,
 				error,
+				num_patterns: searchTerm ? patternsCount : undefined,
 			} );
 		}
 
 		// We want to avoid resubmitting the event whenever
 		// `category` changes, which is why we deliberately don't include it in the dependency array
 		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [ isDevAccount, isLoggedIn, debouncedSearchTerm, patternTypeFilter, view ] );
+	}, [ isDevAccount, isLoggedIn, searchTerm, patternTypeFilter, view ] );
```

Comment:
```suggestion
	}, [ isDevAccount, isLoggedIn, searchTerm, patternsCount, patternTypeFilter, view ] );
```
We need to include this property here as well

Reply:
Hm, could you please elaborate?

What I see:
1) We have `searchTerm`, so its' literally the same check
2) The event will be triggered every time when we do pagination, but I suppose we don't need it triggered in this case

Reply:
> We have searchTerm, so its' literally the same check

`searchTerm` changes independently of `patternsCount`, but if `patternsCount` is `undefined` initially, then we want to run the effect again when it changes.

> The event will be triggered every time when we do pagination

It wouldn't actually, because the pagination logic happens in `PatternGallery`, so it doesn't affect the `patternsCount` that we pass to this component.